internal/abci/abci.go

```diff
@@ -478,10 +501,9 @@ func (a *AbciApp) FinalizeBlock(ctx context.Context, req *abciTypes.RequestFinal
 		res.TxResults = append(res.TxResults, abciRes)
 
 		// Remove the transaction from the cache as it has been included in a block
-		hash := cmtTypes.Tx(tx).Hash()
```

Comment:
Here's another optimization in `FinalizeBlock`: it hashed each transaction twice.